website/server/controllers/api-v3/pushNotifications.js

```diff
@@ -0,0 +1,95 @@
+import { authWithHeaders } from '../../middlewares/api-v3/auth';
+import {
+  BadRequest,
+  NotAuthorized,
+} from '../../libs/api-v3/errors';
+
+let api = {};
+
+/**
+ * @apiIgnore
+ * @api {post} /api/v3/user/add-push-device Add a push device to a user
+ * @apiVersion 3.0.0
+ * @apiName UserAddPushDevice
+ * @apiGroup User
+ *
+ * @apiParam {string} regId The id of the push device
+ * @apiParam {string} type The type of push device
+ *
+ * @apiSuccess {Object} data List of push devices
+ * @apiSuccess {string} message Success message
+ */
+api.addPushDevice = {
+  method: 'POST',
+  url: '/user/push-devices',
+  middlewares: [authWithHeaders()],
+  async handler (req, res) {
+    let user = res.locals.user;
+
+    req.checkBody('regId', res.t('regIdRequired')).notEmpty();
+    req.checkBody('type', res.t('typeRequired')).notEmpty();
+
+    let validationErrors = req.validationErrors();
+    if (validationErrors) throw validationErrors;
+
+    let pushDevices = user.pushDevices;
+
+    let item = {
+      regId: req.body.regId,
+      type: req.body.type,
+    };
+
+    if (pushDevices.find(device => device.regId === item.regId)) {
+      throw new NotAuthorized(res.t('pushDeviceAlreadyAdded'));
+    }
+
+    pushDevices.push(item);
+
+    await user.save();
+
+    res.respond(200, {pushDevices: user.pushDevices}, res.t('pushDeviceAdded'));
```

Comment:
maybe here pass user.pushDevices directly instead of wrapping them inside an object. This way you'll get {data: {pushDevices: []}}, isn't better {data: []}?